fdbserver/VersionedBTree.actor.cpp

```diff
@@ -4134,8 +4134,9 @@ class VersionedBTree : public IVersionedStore {
 
 		state Version writeVersion = self->getLastCommittedVersion() + 1;
 
-		wait(self->m_commitReadLock.take());
-		state FlowLock::Releaser readLock(self->m_commitReadLock);
+		state Reference<FlowLock> commitReadLock = self->m_commitReadLock;
+		wait(commitReadLock->take());
+		state FlowLock::Releaser readLock(*commitReadLock);
```

Comment:
Is this the fix to the failures on the master we have seen yesterday?

Reply:
It fixes one of the Redwood crashes I mentioned earlier.  There's another one of the same form for the read lock.   Even when running 100% ssd-2, release branch is not clean so master probably is not either.